File: RemoveKdigits.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        if(k==num.size())return "0";
        stack < char >  s;
        int i=0, n=num.size();
        while(i < n)
        {
            while( !s.empty() && k > 0 &&s.top()>num[i])
            {
                s.pop();
                k--;
            }
            s.push(num[i]);
            i++;
        }
        while(k>0)
        {
            s.pop();
            k--;
        }
        string ans="";
        while(!s.empty())
        {
            ans=s.top()+ans;
            s.pop();
        }
        while(ans.size()>1&&ans[0]=='0')
            ans.erase(0,1);
        return ans;
    }
};

File: Remove K Digits.py
# Day : 13

class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        while k > 0:
            k -= 1
            i = 0
            while i < len(num)-1:
                if num[i] > num[i+1]:
                    break
                i += 1
            num = num[:i] + num[i+1:]
        
        if len(num) == 0:
            return "0"
        else:
            return str(int(num))
